projects/js-packages/components/components/boost-score-graph/get-date-format.ts

```diff
@@ -0,0 +1,59 @@
+// because 'en' defaults to the OS variant - replace numeric month with short month to avoid ambiguity
+const MONTH_FORMAT = 'short';
+
+/**
+ * Returns a formatted date based on the provided template and locale.
+ *
+ * @param {string} template - The template used to format the date.
+ * @param {Date} date - The date object to be formatted.
+ * @param {string} [locale='en'] - The locale code specifying the language and region to be used for formatting. Default 'en'.
+ * @returns {string} The formatted date as a string.
+ */
+export default function getDateFormat( template: string, date: Date, locale = 'en' ): string {
```

Comment:
> Have I understood the role of the fmtDate function correctly?

@thingalon, yes you are right!

`uPlot.fmtDate`  is the default date formatted. We can overwrite it to display our custom locale instead of the one provided by the browser. We can also receive the format date string from uPlot and output what we want instead, and overwriting the uPlot's decision.

> Where do the date format strings come from? Are they decided by uPlot, or do we supply them somewhere?

uPlot decides what format to display in the cart, if your range is just a day, it will add hours and minutes, if you range is a year, it will display only the month names. So overwritting this function gives us the control to display the part of the date we want.

> If we supply the date strings, can we supply format strings compatible with [WordPress' date functions](https://github.com/WordPress/gutenberg/blob/80f2c2ee0780e90e2bf809b0569f70276fae4a7f/packages/date/README.md) and use those instead of custom code?

I'll review the link. From what I know we cannot define the format string just interpret them.

> If uPlot supplies the format functions, do we know a complete list of templates they might send us? It looks like this switches between a selection of pre-set templates, instead of understanding the {M}, {D}, etc markers. Is that a concern?

Yes, it does supply the format function and the format template. `uPlot.fmtDate` default function is also customizable but less useful for us, because we would need to pass the values for each locale. Here is a demo view-source:https://leeoniya.github.io/uPlot/demos/months-ru.html


I'll take a deeper look this week to see if uPlot provides a better option for our use case.
The original implementation of this function comes from Calypso: https://github.com/Automattic/wp-calypso/blob/360a4c9a8265a9e5abf5037196fc41945ec58f4b/packages/components/src/chart-uplot/lib/get-date-format.ts